core/iwasm/libraries/thread-mgr/thread_manager.c

```diff
@@ -284,6 +284,20 @@ wasm_cluster_del_exec_env(WASMCluster *cluster, WASMExecEnv *exec_env)
 {
     bool ret = true;
     bh_assert(exec_env->cluster == cluster);
+
+#if WASM_ENABLE_DEBUG_INTERP != 0
+    /* Wait for debugger control thread to process the
+       stop event of this thread */
+    if (cluster->debug_inst) {
+        /* lock the debug_inst->wait_lock so
+           other threads can't fire stop events */
+        os_mutex_lock(&cluster->debug_inst->wait_lock);
+        while (cluster->debug_inst->stopped_thread == exec_env) {
```

Comment:
Same as above, can we add os_cond_timedwait(&cluster->debug_inst->wait_cond, ..) inside loop?

Reply:
Done, thanks!